gensim/models/wrappers/ldamallet.py

```diff
@@ -269,7 +289,30 @@ def read_doctopics(self, fname, eps=1e-6, renorm=True):
                            for id_, weight in enumerate(map(float, parts))
                            if abs(weight) > eps]
                 else:
-                    raise RuntimeError("invalid doc topics format at line %i in %s" % (lineno + 1, fname))
+                    if mallet_version == "2.0.7":
+                        """
+
+                            1   1   0   1.0780612802674239  30.005575655428533364   2   0.005575655428533364    1   0.005575655428533364    
+                            2   2   0   0.9184413079632608  40.009062076892971008   3   0.009062076892971008    2   0.009062076892971008    1   0.009062076892971008
+                            In the above example there is a mix of the above if and elif statement. There are neither `2*num_topics` nor `num_topics` elements.
+                            It has 2 formats 40.009062076892971008 and 0   1.0780612802674239 which cannot be handled by above if elif.
+                            Also, there are some topics are missing(meaning that the topic is not there) which is another reason why the above if elif
+                            fails even when the `mallet` produces the right results
+
+                        """
+                        pointer = 0
+                        doc = []
+                        while pointer < len(parts):
+                            if float(parts[pointer]) == int(parts[pointer]):
+                                if float(parts[pointer+1]) > eps:
```

Comment:
Looking at the `while` condition, this `[pointer + 1]` access could result in out-of-bounds `IndexError`.

Also, the logic around these `if`s deserves some comments, it's not clear what's happening and why.